util-ui/src/main/kotlin/com/tailoredapps/androidutil/ui/keyboard/KeyboardStateManager.kt

```diff
@@ -0,0 +1,87 @@
+package com.tailoredapps.androidutil.ui.keyboard
+
+import android.graphics.Rect
+import android.view.View
+import android.view.ViewTreeObserver
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleObserver
+import androidx.lifecycle.OnLifecycleEvent
+import io.reactivex.Observable
+import io.reactivex.disposables.Disposable
+import io.reactivex.functions.Consumer
+import io.reactivex.subjects.BehaviorSubject
+
+/**
+ * Created by alexandergrafl on 2019-11-05.
+ * Manages access to the Android soft keyboard.
+ * Credit to https://medium.com/@munnsthoughts/detecting-if-the-android-keyboard-is-open-using-kotlin-rxjava-2-8aee9fae262c
+ * Use with care: This class keeps a reference to the view/activity!
+ */
+class KeyboardStateManager(private val activity: AppCompatActivity) : LifecycleObserver {
+
+    private val stateSubject = BehaviorSubject.create<KeyboardStatus>()
+
+    private val stateDisposable: Disposable
+
+    init {
```

Comment:
I'm not really sure about the lifecycle-class-coherence, but is ti theoretically be possible that one instanceif the `KeyboardStateManager` is created, then the activity is destroyed, and the same `KeyboardStateManager` instance is then used again? In that case the content of the init here would need to be duplicated in the onCreate lifecycle method.

(I'm kind of thinking about the _rotate Activity_ use case here)